### PR TITLE
Removed superfluous interfaces and dead code

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
The projects have been bloated with a lot of unnecessary interfaces which are abstracting when abstracting is not needed or useful.

# What has changed
The interfaces have been replaced with the concrete implementations.

A class and unit test which weren't being used anymore have been removed.

# How to test?
Have run ITs and ATs to check there's no regression - passing OK.

# Links
Trello: https://trello.com/c/CrZzmiKn/329-get-rid-of-all-the-interfaces-in-the-java-projects

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/sample/message/impl/SFTPSampleReceiver.java [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-    log.debug("Renaming failed for" + filename);
-  }
-
-  /**
-   * Creates error file containing the reason for XML validation failure
-   *
-   * @param errorMessage failure message containing reason for failure
-   * @return Message<String> message containing cut down error message and new file names
-   */
-  @ServiceActivator(inputChannel = "pollerErrorChannel", outputChannel = "errorUploadChannel")
```
Comment (benjefferies) [inline src/main/java/uk/gov/ons/ctp/response/sample/message/impl/SFTPSampleReceiver.java]: Assuming this outputChannel does nothing?

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/sample/message/impl/SFTPSampleReceiver.java]: well, neither the input nor the output channel are declared in any of the springintegration XMLs, so I don't think those rabbit queues/topics even exist

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/sample/message/impl/SFTPSampleReceiver.java]: we know that the sample service receives its files via MIME multipart upload. Perhaps there was - in prehistoric times - a plan to use SFTP as an alternative delivery method??! Call the software historians  ;)

**Comment by NickGrantONS** [comment]
the JAR isn't used by other projects so I don't know why we'd need javadocs - this isn't an API/library project

**Pull request merged**